apps/docs/content/5.extend/8.custom-drains.md

```
@@ -0,0 +1,339 @@
+    to: /reference/best-practices
+    color: neutral
+    variant: subtle
+---
+
+A **drain** is the terminal step of evlog's pipeline: a function that receives wide events and ships them somewhere — an HTTP API, a message queue, a database, a webhook, a local file. evlog ships built-in drains for popular providers ([Adapters overview](/integrate/adapters/overview)). When you need a destination that isn't covered, you write your own.
+
+Two factories cover every case:
+
+| You have… | Use |
+| --- | --- |
+| An HTTP backend (REST, JSON ingest, vendor `/v1/logs` endpoint) | [`defineHttpDrain`](#definehttpdrain-the-http-recipe) |
```

markdownlint-cli2 (0.22.1)

[warning] Link fragments should be valid

(MD051, link-fragments)